pos_order_tax_detail/models/pos_order.py

```diff
@@ -0,0 +1,80 @@
+#
+#    OpenERP, Open Source Management Solution
+#    This module copyright :
+#        (c) 2015 Antiun Ingenieria, SL (Madrid, Spain, http://www.antiun.com)
+#                 Antonio Espinosa <antonioea@antiun.com>
+#
+#    This program is free software: you can redistribute it and/or modify
+#    it under the terms of the GNU Affero General Public License as
+#    published by the Free Software Foundation, either version 3 of the
+#    License, or (at your option) any later version.
+#
+#    This program is distributed in the hope that it will be useful,
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import models, fields, api
+import logging
+
+_logger = logging.getLogger(__name__)
+
+
+class PosOrder(models.Model):
+    _inherit = 'pos.order'
+
+    taxes = fields.One2many(comodel_name='pos.order.tax',
+                            inverse_name='pos_order')
+
+    @api.multi
+    def compute_tax_detail(self):
+        for order in self:
+            taxes_to_delete = self.env['pos.order.tax'].search(
+                [('pos_order', '=', order.id)])
+            if order.lines:
+                taxes = order.lines.taxes_get()
+                for key, tax in taxes.iteritems():
+                    current = taxes_to_delete.filtered(
+                        lambda r: r.tax.id == tax['id'])
+                    if current:
+                        current.write({
+                            'base': tax['base'],
+                            'amount': tax['amount'],
+                        })
+                        taxes_to_delete -= current
+                    else:
+                        self.env['pos.order.tax'].create({
+                            'pos_order': order.id,
+                            'tax': tax['id'],
+                            'name': tax['name'],
+                            'base': tax['base'],
+                            'amount': tax['amount'],
+                        })
+
+            if taxes_to_delete:
+                taxes_to_delete.unlink()
+        return True
+
+    @api.multi
+    def action_paid(self):
+        super(PosOrder, self).action_paid()
+        self.compute_tax_detail()
+        return True
+
+    @api.model
+    def _install_tax_detail(self):
+        """Create tax details to pos.order's already paid, done or invoiced.
+        """
+        # Find orders with state : paid, done or invoiced
+        orders = self.search([('state', 'in', ('paid', 'done', 'invoiced'))])
+        # Compute tax detail
+        orders.compute_tax_detail()
```

Comment:
@legalsylvain, thanks for your warning

Taxes are copied (name, base, amount) when order is paid, so taxes are the ones were configured in product when order was paid. Taxes can be changed in product, but this will not affect to order.

Original Odoo code does the same, calculates amount_tax and amount first in JS code (POS) and then recalculates taxes again when creating account_move when POS session is closing. So, you can check this in runbot and see that payments are incoherent with account moves created when POS session is closed and validated. 

It's not expected that user change taxes of a product when a POS session is in process, but you're right, I've added this notice in README. I've also implement an amount check to advice user when taxes have been change to one or more products during POS session.